lib/tools/common/patching_parallel.py

```
@@ -0,0 +1,1386 @@
+    groups = list(groups_dict.values())
+
+    # Step 4: Sort each group to preserve original order
+    for group in groups:
+        group.sort(key=lambda p: patches.index(p))
+
+    # Log grouping results
+    total_patches = len(patches)
+    total_groups = len(groups)
+
+    # DEBUG: Log group composition to verify grouping is working
+    log.debug(f"File-based dependency: Sample group composition:")
```

Ruff (0.15.21)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)